presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -241,7 +248,7 @@ private SubPlan buildFragment(PlanNode root, FragmentProperties properties, Plan
             boolean equals = properties.getPartitionedSources().equals(ImmutableSet.copyOf(schedulingOrder));
             checkArgument(equals, "Expected scheduling order (%s) to contain an entry for all partitioned sources (%s)", schedulingOrder, properties.getPartitionedSources());
 
-            Map<Symbol, Type> symbols = Maps.filterKeys(types.allTypes(), in(dependencies));
+            Map<Symbol, Type> symbols = Maps.filterKeys(symbolAllocator.getTypes().allTypes(), in(dependencies));
```

Comment:
why?

Reply:
@rschlussel : Because now we can directly use the `types` from `symbolAllocator` ? -- previously we need callee to pass in ? :)